internal/server/singleprocess/state/workspace.go

```diff
@@ -14,16 +14,43 @@ import (
 	pb "github.com/hashicorp/waypoint/internal/server/gen"
 )
 
-var (
-	workspaceBucket = []byte("workspaces")
-)
+var workspaceBucket = []byte("workspaces")
 
 func init() {
 	dbBuckets = append(dbBuckets, workspaceBucket)
 	dbIndexers = append(dbIndexers, (*State).workspaceIndexInit)
 	schemas = append(schemas, workspaceIndexSchema)
 }
 
+// WorkspacePut creates or updates the given Workspace.
+//
+// Project changes will be ignored
+func (s *State) WorkspacePut(workspace *pb.Workspace) error {
+	memTxn := s.inmem.Txn(true)
+	defer memTxn.Abort()
+
+	err := s.db.Update(func(dbTxn *bolt.Tx) error {
+		prev, err := s.workspaceGet(dbTxn, memTxn, &pb.Ref_Workspace{
```

Comment:
I think this logic also should exist in the workspacePut.

Reply:
Moved in e025389f8